Sindri-Labs/sindri-resources
# Adding Halo2 pse-v0.3.0 Samples

This adds two circuits demonstrating our support and requirements for Halo2 via the PSE fork (v0.3.0).  The simpler circuit computes the dot product of two vectors, while the larger circuit was constructed from an external (highly compatible) codebase.  See the readme for the merkle tree circuit for more details.  

## Testing Instructions
Submit both circuits to our frontend to confirm that they compile 
- [x] `circuit_database/halo2/pse-v0.3.0/merkle_tree.zip`
- [x] `circuit_database/halo2/pse-v0.3.0/vector_multiplication.zip`

Notes
* the input JSON files are valid.  You can use the Sindri CLI to create a proof from these circuits if you'd like to see the proving output
* lint is failing because our Sindri manifest schema has not been updated to allow the new PSE halo2 subtype.  (This is a non-issue)
